docs/index.rst

```diff
@@ -21,7 +21,6 @@ Contents
   overview_and_guide
   api_walkthrough
   rest_api
-  what_is_the_dev_preview
```

Comment:
I'd actually keep the doc itself in the repo, but just not link to it. Just for historical reasons. But I don't have a strong opinion.

Reply:
Fine for me, but we can always restore it from git if necessary.

Reply:
Let's remove it. It's in git.